python_modules/dagster/dagster/_core/definitions/partition.py

```diff
@@ -1386,6 +1445,22 @@ def from_serialized(
                 )
             return cls(subset=set(data.get("subset")), partitions_def=partitions_def)
 
+    @classmethod
+    def can_deserialize(
+        cls,
+        partitions_def: PartitionsDefinition,
+        serialized: str,
+        serialized_partitions_def_unique_id: Optional[str],
+        subset_type: Optional[PartitionsSubsetType],
+    ) -> bool:
+        if subset_type is not None:
+            return subset_type == cls.subset_type()
+
+        data = json.loads(serialized)
```

Comment:
Does this mean that we'd now be deserializing the data twice when previously we were doing it once? I think this deserialization can be a pretty expensive operation in some situations, so ideally we'd just do it once.

Reply:
Yes, but in the future when all backfills have the subset type and serializable partitions def unique ID, we won't need to deserialize the data in order to check the validity. We only need to deserialize for the backcompat cases where we don't have the subset type / unique ID.

I think that the number of backcompat cases shouldn't be excessively high given that we only added serialized asset backfill data a month ago.

Reply:
Ah I see - just the backcompat cases seems fine